cli/fabric.c

```diff
@@ -46,14 +66,14 @@ enum {
 static int raw_print_hex(unsigned long long value,
 			 int offset, int bytes)
 {
-	printf("%06X - 0x%0*llX\n", offset, bytes * 2, value);
+	printf("%06X - 0x%0*" PFMTllX "\n", offset, bytes * 2, value);
```

Comment:
You should probably tell us what the build warnings are... but this all looks really wrong....

'llx' should always be correct for any long long value.... PRIx64 should be used for uint64_t and 'value' is a long long...

Reply:
The warnings was due to difference in Windows, e.g.,:
- Format `%llx` is not recognized on Windows. On Windows it should be `I64x`. These are now replaced with `FMT_llx` macro, which expands to the right format for the platform.
- Format `%hhu` is not recognized on Windows. Replaced with `PRIu8` which expands to the right format.
...

Reply:
According to this they should be recognized... 

https://docs.microsoft.com/en-us/cpp/c-runtime-library/format-specification-syntax-printf-and-wprintf-functions?view=vs-2019

And when I try a simple program with x86_64-w64-mingw32-gcc it recognizes the %ll and %hh just fine...

The actual warning messages you are seeing should really be in the commit message even if this patch was obviously correct.

Also, if there were build warnings in the code (even for windows), Travis should have caught them and prevented the code from being merged. So are you trying to do the windows build on some ancient compiler or something like that? This info needs to be in the commit message.

Reply:
Our code is compiled with `-Wall` so I got these warnings:
```
cli/fabric.c: In function 'raw_print_hex':
cli/fabric.c:49:23: warning: unknown conversion type character 'l' in format [-Wformat=]
  printf("%06X - 0x%0*llX\n", offset, bytes * 2, value);
                       ^
cli/fabric.c:49:9: warning: too many arguments for format [-Wformat-extra-args]
  printf("%06X - 0x%0*llX\n", offset, bytes * 2, value);
...
```


I compiled the following code with `-Wall` using x86_64-w64-mingw32-gcc (ver. 8.3.0)and got the same warnings.
```
#include "stdio.h"
int main()
{
        unsigned long long x = 0x1234567898765432;
        printf("%llx\n", x);
        return 0;
}
```

The code does print correctly, so it's just the warning from the compiler that we need to worry about.

For Travis, the warnings are there in the logs, however, they are not considered as build errors. So in the summary we see `build pass`.

Reply:
Ah I see...

How about instead of this mess we set __USE_MINGW_ANSI_STDIO per this [document][1]? I'd just set it on the command line for windows (ie -D__USE_MINGW_ANSI_STDIO).

Also, we should probably ensure Travis is running with -Werror so we fix warnings before they enter the source.

[1]: https://sourceforge.net/p/mingw-w64/wiki2/gnu%20printf/

Reply:
The macro -D__USE_MINGW_ANSI_STDIO does work for `fabric.c`, but `portable.h` has the following definitions:
```
#ifdef __WINDOWS__
#define FMT_SIZE_T_x  "Ix"
#define FMT_llX       "I64X"
#define FMT_llx       "I64x"
#define FMT_lld       "I64d"
```
So the files that uses these defines (gas.c, etc.) failed to build. 

There are 2 paths we can take from here:
- remove definitions from `portable.h` and fix `gas.c`, use __USE_MINGW_ANSI_STDIO for all files
- keep `portable.h` definitions and use current build fix

I think it's better to keep `portable.h` definitions so the code has better portability to other platform (with different definitions for `llx`).

What do you think? @lsgunth

Reply:
I think it's better to remove the portable.h junk so the code is much cleaner.

There are no platforms that have different definitions of llx. Only ancient windows platforms have that issue and we can substitute in a good version for all cases with the flag above.

Reply:
Updated based on your suggestions. Please take a look. @lsgunth